traitcuration/traits/datasources/ols.py

```diff
@@ -45,3 +49,54 @@ def get_ontology_id(term_iri):
     if ontology_id == 'orphanet':
         ontology_id = 'ordo'
     return ontology_id
+
+
+def get_term_status(is_obsolete, ontology_id=None):
+    """
+    Takes the ontology_id of a term and a whether it is obsolete or not, and returns its calculated status.
+    'Obsolete' if the is_obsolete flag is true, 'Deleted' if no info was found in any ontology, 'Current' if its
+    ontology is EFO, and 'Needs Import' if info about the term was found in another ontology.
+    """
+    if ontology_id is None:
+        return Status.DELETED
+    if is_obsolete:
+        return Status.OBSOLETE
+    if ontology_id == 'efo':
+        return Status.CURRENT
+    return Status.NEEDS_IMPORT
```

Comment:
The overall logic of status selection & querying OLS in a single function is now much better. However, what worries me is that we have two completely separate functions which decide the statuses: `get_term_status` (which is, by the way, defined in _ols.py_ but used only in _zooma.py_ — looks a bit odd) and `update_status`. So ultimately there are two places in the code where the term can be decided to be deleted or obsolete. Ideally, this should be done only once, and all of the different cases must use exactly the same status determination logic:
* Term is added from ZOOMA
* Term is added by user
* Term status is being updated by periodic OLS lookup

Reply:
Yes, I agree, I will rethink that and merge those two function into a single calculation